content/docs/document-lifecycle.md

```diff
@@ -0,0 +1,60 @@
+---
+title: 'Document lifecycle'
+weight: 300
+---
+
+A `LangiumDocument` goes through seven different states during its lifecycle:
+1. `Parsed` when an AST has been generated from the content of the document.
+2. `IndexedContent` when the AST nodes have been processed by the `IndexManager`.
+3. `Processed` when the pre-processing steps have been completed.
+4. `Linked` when the `Linker` has resolved cross-references.
+5. `IndexedReferences` when the references have been indexed by the `IndexManager`.
+6. `Validated` when the document has been validated by the `DocumentValidator`.
+7. `Changed` when the document has been modified.
+
+The following diagram depicts the services involved in the lifecycle as well as their main responsibilities. More details about the lifecycle can be found below.
+
+![Lifecycle diagram](/assets/doc-lifecycle/document-lifecycle.png)
```

Comment:
The white background doesn't look very good with dark theme. We should consider disabling theme switching and always show dark theme to be consistent with the landing page. In this case, the image colors should be inverted.

Alternatively, we could at least try to use lower-contrast colors in the diagram so its fits better with both themes.